contracts/BondStorage.sol

```diff
@@ -18,13 +19,15 @@ contract BondStorage is AccessControl {
     uint8 public otherUsdDec;
 
     constructor(address _gatewayAddress, address _chainlinkEurOther, uint8 _otherUsdDec) {
-        _setupRole(WHITELIST_BOND_STORAGE, msg.sender);
+        _grantRole(WHITELIST_ADMIN, msg.sender);
+        _setRoleAdmin(WHITELIST_BOND_STORAGE, WHITELIST_ADMIN);
+        grantRole(WHITELIST_BOND_STORAGE, msg.sender);
         tokenGateway = StandardTokenGateway(_gatewayAddress);
         chainlinkEurOther = _chainlinkEurOther;
         otherUsdDec = _otherUsdDec;
     }
 
-    modifier onlyOwner() {
+    modifier onlyWhitelisted() {
         require(hasRole(WHITELIST_BOND_STORAGE, msg.sender), "invalid-user");
```

Comment:
we have many error messages with `invalid-user` now, we should start to be more specific

Reply:
that's a good idea 👍🏻